src/link.rs

```diff
@@ -25,6 +25,20 @@ pub struct Properties {
     pub title: String,
     pub desc: Option<String>,
 }
+/// Write data to a tempory file and move that written file to destination
+///
+/// May failed if writing or moving failed
+fn temp_and_move(
```

Comment:
Probably we should expose this method so apps could use it directly (e.g. Retouch or Memo).

Reply:
We should consider making "atomic files" API public, too. But not sure about the use-case yet.